Trend Micro/trend-micro-apex-one/ingest/parser.yml

```
@@ -0,0 +1,573 @@
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          file.name: "{{parsed_event.message.fname}}"
+          url.original: "{{parsed_event.message.request}}"
+          Trend_Micro.Security_Threat: "{{parsed_event.message.cs1}}"
+          Trend_Micro.Risk_Level: "{{parsed_event.message.cn1}}"
+          Trend_Micro.Threat_Categories: "{{parsed_event.message.cs2}}"
+          Trend_Micro.Cloud_Service_Vendor: "{{parsed_event.message.cs3}}"
+
+  set_spyware_grayware_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          Trend_Micro.VirusName: "{{parsed_event.message.cs1}}"
+          file.name: "{{parsed_event.message.fname}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          host.name: "{{parsed_event.message.dhost}}"
+          source.ip: "{{parsed_event.message.dst or parsed_event.message.c6a3}}"
+          user.name: "{{parsed_event.message.duser}}"
+
+  set_suspicious_files_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          file.name: "{{parsed_event.message.cs3}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          host.name: "{{parsed_event.message.dhost}}"
+          source.ip: "{{parsed_event.message.dst or parsed_event.message.c6a3}}"
+          event.action: "{{parsed_event.message.act}}"
+
+  set_virus_malware_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["malware"]
+          event.type: ["info"]
+          host.name: "{{parsed_event.message.dhost}}"
+          user.name: "{{parsed_event.message.duser}}"
+          event.code: "{{parsed_event.message.cs4}}"
+          file.name: "{{parsed_event.message.fname}}"
+          file.path: "{{parsed_event.message.filePath}}"
+          file.hash.sha1: "{{parsed_event.message.fileHash}}"
+          source.ip: "{{parsed_event.message.dst or parsed_event.message.c6a3}}"
+
+  set_web_security_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["network"]
+          event.type: ["info"]
+          destination.port: "{{parsed_event.message.dpt}}"
+          source.ip: "{{parsed_event.message.src}}"
+          rule.ruleset: "{{parsed_event.message.cs1}}"
+          host.name: "{{parsed_event.message.dvchost}}"
+          file.name: "{{parsed_event.message.fname}}"
+          url.original: "{{parsed_event.message.request}}"
+          user.name: "{{parsed_event.message.duser}}"
+          destination.ip: "{{parsed_event.message.dst}}"
+
+      - translate:
+        dictionary:
+          0: Unknown
+          1: Pass
+          2: Block
+          3: Monitor
+          4: Delete
+          5: Quarantine
+          6: Warn
+          7: Warn and continue
+          8: Override
+        mapping:
+          parsed_event.message.act: event.action
+
```

I suggest defining `event.type` according to the action
```suggestion

      - translate:
        dictionary:
          1: ["allowed"]
          2: ["denied"]
          4: ["denied"]
        mapping:
          parsed_event.message.act: event.type
        fallback: ["info"]
          
```